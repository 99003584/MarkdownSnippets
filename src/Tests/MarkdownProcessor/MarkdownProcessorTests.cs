using System.Collections.Generic;
using System.IO;
using MarkdownSnippets;
using Xunit;
using Xunit.Abstractions;

public class MarkdownProcessorTests :
    TestBase
{
    [Fact]
    public void Toc()
    {
        var markdownContent = @"
# Title

toc

## Heading 1

Text1

## Heading 1

Text2

";
        SnippetVerifier.Verify(markdownContent,new List<Snippet>(),new List<string>());
    }

    [Fact]
    public void Simple()
    {
        var availableSnippets = new List<Snippet>
        {
            SnippetBuild(
                language: "cs",
                key: "snippet1"
            ),
            SnippetBuild(
                language: "cs",
                key: "snippet2"
            )
        };
        var markdownContent = @"
snippet: snippet1

some text

snippet: snippet2

some other text

snippet: FileToUseAsSnippet.txt

some other text

snippet: /FileToUseAsSnippet.txt

";
        SnippetVerifier.Verify(
            markdownContent,
            availableSnippets,
            new List<string>
            {
                Path.Combine(GitRepoDirectoryFinder.FindForFilePath(), "src/Tests/FileToUseAsSnippet.txt")
            });
    }

    static Snippet SnippetBuild(string language, string key)
    {
        return Snippet.Build(
            language: language,
            startLine: 1,
            endLine: 2,
            value: "Snippet",
            key: key,
            path: "thePath");
    }

    public MarkdownProcessorTests(ITestOutputHelper output) :
        base(output)
    {
    }
}