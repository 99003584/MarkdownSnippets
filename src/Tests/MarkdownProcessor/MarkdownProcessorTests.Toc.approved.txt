{
  MissingSnippets: [],
  UsedSnippets: [],
  content: '<!--
GENERATED FILE - DO NOT EDIT
This file was generated by [MarkdownSnippets](https://github.com/SimonCropp/MarkdownSnippets).
To change this file edit the source file and then run MarkdownSnippets.
-->


# Title

<!-- toc -->
## Contents

 * [Heading 1](#heading-1)
 * [Heading 1](#heading-1)


## Heading 1

Text1

## Heading 1

Text2

'
}